src/Nethermind/Nethermind.Api/IApiWithBlockchain.cs

```diff
@@ -96,5 +97,7 @@ public interface IApiWithBlockchain : IApiWithStores, IBlockchainBridgeFactory
         IGasPriceOracle? GasPriceOracle { get; set; }
         
         IEthSyncingInfo? EthSyncingInfo { get; set; }
+
+        CompositePruningTrigger PruningTrigger { get; }
```

Comment:
this should be an interface

Reply:
The idea here is that there can be multiple triggers, but we need a central place to collect them, hence the class that can handle multiple interface items via Add.